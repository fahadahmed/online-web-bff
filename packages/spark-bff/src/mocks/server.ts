import { setupServer } from 'msw/node';
import a2pCustomerHandlers from 'datasources/a2p/customer/mocks/handlers';
import accessHandlers from 'datasources/user/access/mocks/handlers';
import accountSummaryHandlers from 'datasources/customer/accountsService/mocks/handlers';
import addressHandlers from 'datasources/location/address/mocks/handlers';
import airpointsHandlers from 'datasources/airpoints/mocks/handlers';
import availableNumbers from 'datasources/shop/availableNumbers/mocks/handlers';
import balanceServiceHandlers from 'datasources/finance/balanceService/mocks/handlers';
import billPaymentHandlers from 'datasources/finance/paymentService/billPayment/mocks/handlers';
import cartServicesHandlers from 'datasources/shop/cart/mocks/handlers';
import checkoutServicesHandlers from 'datasources/shop/checkout/mocks/handlers';
import connectionDiagnosticsHandlers from 'datasources/line/assure/connectionDiagnostics/mocks/handlers';
import connectionPromiseHandler from 'datasources/line/assure/connectionPromise/mocks/handlers';
import contentServiceHandlers from 'datasources/utility/content/mocks/handlers';
import digitalBillHandlers from 'datasources/finance/digitalBill/mocks/handlers';
import identityServiceHandlers from 'datasources/user/identityService/mocks/handlers';
import lineConfiguarationHandlers from 'datasources/line/lineConfiguration/mocks/handlers';
import lineDetailServiceHandlers from 'datasources/line/lineDetails/mocks/handlers';
import lineDetailsHandlers from 'datasources/line/lineSummary/mocks/handlers';
import lineSpendHistoryHandlers from 'datasources/line/spend/mocks/handlers';
import lineSubscriptionsService from 'datasources/line/lineSubscriptionsService/mocks/handlers';
import lineUsageDetailsMocks from 'datasources/line/usage/usageDetailsService/mocks';
import lineUsageHistoryMocks from 'datasources/line/usage/usageHistoryService/mocks';
import lineUsageMocks from 'datasources/line/usage/mocks/handlers';
import lineUsageTransactionMocks from 'datasources/line/usage/usageTransactionsService/mocks';
import networkSettingsHandlers from 'datasources/technology/networkSettingsService/mocks/handlers';
import orderHandlers from 'datasources/shop/order/mocks/handlers';
import paymentServiceHandlers from 'datasources/finance/paymentService/mocks/handlers';
import pointOfAddressHandlers from 'datasources/location/pointOfInterest/mocks/handlers';
import pointOfOutageHandlers from 'datasources/location/pointOfInterest/outagesPointOfInterest/mocks/handlers';
import preferenceHandlers from 'datasources/user/preference/mocks/handlers';
import prepaidBalanceServiceHandlers from 'datasources/finance/prepaidBalanceService/mocks/handlers';
import productDetailServiceHandlers from 'datasources/product/details/mocks/handlers';
import productFiltersServiceHandlers from 'datasources/product/filters/mocks/handlers';
import productOffersHandlers from 'datasources/product/offers/mocks/handlers';
import productPromotionsMPDServiceHandlers from 'datasources/product/promotions/mpd/mocks/handlers';
import profileServiceHandlers from 'datasources/marketing/profileService/mocks/handlers';
import resourceHandlers from 'datasources/utility/resource/mocks/handlers';
import roamingRatesHandlers from 'datasources/product/roamingRates/mocks/handlers';
import subscriptionActivationHandlers from 'datasources/line/subscriptionActivationService/mocks/handlers';
import topupHandlers from 'datasources/finance/paymentService/topup/mocks/handlers';
import transactionsListHandlers from 'datasources/line/transactionsService/mocks/handlers';
import troubleshootHandlers from 'datasources/line/assure/connectionTroubleshooter/mocks/handlers';
import userInfoHandlers from 'datasources/user/userInfo/mocks/handlers';
import walletServiceHandlers from 'datasources/finance/walletService/mocks/handlers';

const handlers = [
  ...a2pCustomerHandlers,
  ...accessHandlers,
  ...accountSummaryHandlers,
  ...addressHandlers,
  ...availableNumbers,
  ...airpointsHandlers,
  ...balanceServiceHandlers,
  ...billPaymentHandlers,
  ...cartServicesHandlers,
  ...checkoutServicesHandlers,
  ...connectionDiagnosticsHandlers,
  ...connectionPromiseHandler,
  ...contentServiceHandlers,
  ...digitalBillHandlers,
  ...identityServiceHandlers,
  ...lineConfiguarationHandlers,
  ...lineDetailServiceHandlers,
  ...lineDetailsHandlers,
  ...lineSpendHistoryHandlers,
  ...lineSubscriptionsService,
  ...lineUsageDetailsMocks,
  ...lineUsageHistoryMocks,
  ...lineUsageMocks,
  ...lineUsageTransactionMocks,
  ...networkSettingsHandlers,
  ...orderHandlers,
  ...paymentServiceHandlers,
  ...pointOfAddressHandlers,
  ...pointOfOutageHandlers,
  ...preferenceHandlers,
  ...prepaidBalanceServiceHandlers,
  ...productDetailServiceHandlers,
  ...productFiltersServiceHandlers,
  ...productOffersHandlers,
  ...productPromotionsMPDServiceHandlers,
  ...profileServiceHandlers,
  ...resourceHandlers,
  ...roamingRatesHandlers,
  ...subscriptionActivationHandlers,
  ...topupHandlers,
  ...transactionsListHandlers,
  ...troubleshootHandlers,
  ...userInfoHandlers,
  ...walletServiceHandlers,
];

const server = setupServer(...handlers);
export default server;
