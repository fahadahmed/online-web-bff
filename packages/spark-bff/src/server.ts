import { ApolloServer } from 'apollo-server-fastify';
import {
  ApolloServerPluginLandingPageDisabled,
  ApolloServerPluginLandingPageGraphQLPlayground,
} from 'apollo-server-core';
import { logger } from 'apolloLogger';
import context from 'context';
import dataSources from 'datasources';
import { ApolloLoggerPlugin, ApplicationLogEvent } from 'logging';
import schema from 'schema';
import { formatResponse } from 'utils/formatResponse';
import { isDevelopmentEnvironment } from 'utils/isDevelopmentEnvironment';

process.on('uncaughtException', (err) => {
  logger.error(new ApplicationLogEvent({ message: err.message, context: {} }));
  throw err;
});

const apolloServer = new ApolloServer({
  schema,
  dataSources,
  formatResponse,
  logger,
  context,
  introspection: isDevelopmentEnvironment(),
  plugins: [
    new ApolloLoggerPlugin(
      `BFF Server Started. BUILD_ID: ${process.env.BUILD_BUILDID}`,
    ),
    isDevelopmentEnvironment()
      ? ApolloServerPluginLandingPageGraphQLPlayground()
      : ApolloServerPluginLandingPageDisabled(),
  ],
});

export default apolloServer;
