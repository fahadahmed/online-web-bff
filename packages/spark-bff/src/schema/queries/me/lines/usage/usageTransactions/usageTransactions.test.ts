import { createTestClient, gql } from 'test/apolloTestUtils';
import { server } from 'test/server';
import accessMocks from 'datasources/user/access/mocks/handlers';
import usageTransactionMocks from 'datasources/line/usage/usageTransactionsService/mocks';

beforeEach(() => {
  server.use(...accessMocks, ...usageTransactionMocks);
});

test('returns all the Usage Transaction when parameters for line Number of line', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          line(lineNumber: "033227715") {
            usageTransactions(
              start: "2020-01-01T00:00:00.000+13:00"
              end: "2020-03-02T00:00:00.000+13:00"
              size: "20"
            ) {
              transactions {
                transactionType
                startDateTime
                endDateTime
                previousBalance
                currentBalance
                description
                value
                type
                isDebit
              }
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line).toMatchSnapshot();
});
