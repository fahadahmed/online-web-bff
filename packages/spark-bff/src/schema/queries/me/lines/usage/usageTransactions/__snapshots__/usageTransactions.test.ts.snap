// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns all the Usage Transaction when parameters for line Number of line 1`] = `
Object {
  "usageTransactions": Object {
    "transactions": Array [
      Object {
        "currentBalance": 32,
        "description": "$30.00 POLi payment",
        "endDateTime": "2020-02-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-02-01T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 50,
      },
      Object {
        "currentBalance": 100,
        "description": "$30.00 POLi payment",
        "endDateTime": "2020-02-10T23:32:31.000+13:00",
        "isDebit": true,
        "previousBalance": 10,
        "startDateTime": "2020-02-02T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 32,
        "description": "$30.00 POLi payment",
        "endDateTime": "2020-02-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-02-03T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 32,
        "description": "$90.00 POLi payment",
        "endDateTime": "2020-09-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-07-04T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 90,
      },
      Object {
        "currentBalance": 57,
        "description": "$70.00 POLi payment",
        "endDateTime": "2020-04-10T23:32:31.000+13:00",
        "isDebit": true,
        "previousBalance": 2,
        "startDateTime": "2020-03-05T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 57,
        "description": "$70.00 POLi payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": true,
        "previousBalance": 2,
        "startDateTime": "2020-09-06T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 57,
        "description": "$15.00 Poli payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": true,
        "previousBalance": 2,
        "startDateTime": "2020-03-07T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 165,
        "description": "$70.00 POLi payment",
        "endDateTime": "2021-12-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-12-08T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 65,
        "description": "$50.00 POLi payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": true,
        "previousBalance": 2,
        "startDateTime": "2020-09-09T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 57,
        "description": "$70.00 POLi payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": true,
        "previousBalance": 2,
        "startDateTime": "2020-09-10T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 75,
        "description": "$100.00 ANZ payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-09-11T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 86,
        "description": "$70.00 POLi payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-09-12T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 86,
        "description": "$70.00 POLi payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-09-13T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 86,
        "description": "$70.00 POLi payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-09-14T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 86,
        "description": "$70.00 POLi payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-09-15T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 86,
        "description": "$70.00 POLi payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-09-16T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
      Object {
        "currentBalance": 86,
        "description": "$70.00 POLi payment",
        "endDateTime": "2020-12-10T23:32:31.000+13:00",
        "isDebit": false,
        "previousBalance": 2,
        "startDateTime": "2020-09-17T23:32:31.000+13:00",
        "transactionType": "PAYMENT",
        "type": "Payment",
        "value": 30,
      },
    ],
  },
}
`;
